typing/types.ml

```diff
@@ -795,7 +795,7 @@ let set_level ty level =
 (* TODO: introduce a guard and rename it to set_higher_scope? *)
 let set_scope ty scope =
   let ty = repr ty in
-  let prev_scope = ty.scope land marks_mask in
+  let prev_scope = ty.scope land scope_mask in
```

Comment:
Oops. Thanks for the fix.
I suppose that the only consequence was that it would log the new scope every time it is called, so this went undetected, but this is bad enough.

Reply:
Because `prev_scope` was incorrect (it was always 0 for valid calls), we should also be able to observe incorrect behavior after backtracking, where a pre-snapshot scope is (incorrectly) reset to 0.